src/Crate/PDO/PDOStatement.php

```diff
@@ -274,13 +274,20 @@ public function bindParam(
         $length = null,
         $driver_options = null
     ) {
-        if (array_key_exists($parameter, $this->namedToPositionalMap)) {
-            $parameter = $this->namedToPositionalMap[$parameter];
-        } else {
+        if (is_numeric($parameter)) {
             if ($parameter == 0) {
                 throw new Exception\UnsupportedException("0-based parameter binding not supported, use 1-based");
             }
             $parameter--;
+        } else {
+            $namedParameterKey = substr($parameter, 0, 1) == ':' ? substr($parameter, 1) : $parameter;
```

Comment:
@caleblloyd it's a general suggestion - would be picked up by typical IDE CS checks as well